src/chocolatey.resources/helpers/functions/Start-ChocolateyProcessAsAdmin.ps1

```diff
@@ -116,6 +116,76 @@ Elevating Permissions and running [`"$exeToRun`" $wrappedStatements]. This may t
   if ($process.StartInfo.RedirectStandardError) { $process.BeginErrorReadLine() }
   $process.WaitForExit()  
 
+  # This handles installers which exit immediately, and spawn lingering children tasks
+  # which may be carrying on additional installation tasks.
+  #
+  # Few installers spawn child processes which outlive the parent process. So,
+  # let's monitor the exit of these children (and any of their children).
+  #
+  # Implementation inspired by:
+  # http://powershell.com/cs/blogs/tobias/archive/2012/05/09/managing-child-processes.aspx
+  #
+  # TODO: Will some installers provide meaningful exit-codes here?
+  #
+  # BEGIN handling of process children
+    function Get-ProcessChildren {
+    param(
+      $Id
+    )
+
+      Write-Debug "Searching for any active children to process ($($Id))"
+
+      $children = Get-WmiObject -Class Win32_Process -Filter "ParentProcessID=$($Id)"

```

Comment:
Interesting. Is there any issues with using this? I would almost want to wrap this in a try catch, so that if it fails it doesn't cause any issues.